common/web/types/src/util/common-events.ts

```diff
@@ -0,0 +1,21 @@
+import { CompilerErrorNamespace, CompilerErrorSeverity, CompilerEvent } from './compiler-interfaces.js';
+import { ErrorObject } from "ajv";
+
+const m = (code: number, message: string) : CompilerEvent => { return { code, message } };
+
+const KmcErrMask   = CompilerErrorNamespace.Types;
+// const SevInfo = CompilerErrorSeverity.Info | KmcErrMask;
+// const SevHint = CompilerErrorSeverity.Hint | KmcErrMask;
+// const SevWarn = CompilerErrorSeverity.Warn | KmcErrMask;
+// const SevError = CompilerErrorSeverity.Error | KmcErrMask;
+const SevFatal = CompilerErrorSeverity.Fatal | KmcErrMask;
+
+export class CommonTypesMessages {
+  // structured Ajv validation error
+  static Fatal_AJVError = (o:ErrorObject) => m(this.FATAL_XMLValidationError,
+    `${o.instancePath}: ${o.keyword}: ${o.message} ${JSON.stringify(o.params||{})}`);
+  // some other unknown Ajv error. Same as above, but without structure
+  static Fatal_UnknownXMLValidationError = (o:{errorsText: string}) => m(this.FATAL_XMLValidationError,
+    `Validation failed: ${o.errorsText}`);
+  static FATAL_XMLValidationError = SevFatal | 0x0001;
```

Comment:
A few things:

1. `${JSON.stringify(o.params||{})}` needs to be in the caller, to keep the message constants translateable (complex embedded expressions are going to cause grief for translators).
2. These should be ERRORs, not FATALs -- FATALs are reserved for compiler bugs and out of memory.
4. Each error string should be paired with a unique error code -- even though these two errors come from the same code area, we are making this a pattern for ease of maintenance and consistency.
5. The error message needs to be interpretable by a keyboard developer. I'm unclear what is going to be returned by AJV for some of these fields?

```suggestion
  static Error_AJVError = (o:{instancePath:string, keyword:string, message: string, params: string}) => m(this.ERROR_AJVError,
    `Error validating LDML XML file: ${o.instancePath}: ${o.keyword}: ${o.message} ${o.params}`);
  static ERROR_AJVError = SevError | 0x0001;

  // some other unknown Ajv error. Same as above, but without structure
  static Error_UnknownXMLValidationError = (o:{errorsText: string}) => m(this.ERROR_UnknownXMLValidationError,
    `Unknown error validating LDML XML file: ${o.errorsText}`);
  static ERROR_UnknownXMLValidationError = SevError | 0x0002;
```

Reply:
1. good point. 

2. +1

3. makes sense. but related to point 4

4. "what is going to be returned by AJV?"  

There is https://github.com/ajv-validator/ajv-errors which can go really deep with customizing ajv's errors, probably more than we need here.  

The value of `params` is specified [here](https://ajv.js.org/api.html#error-objects) - looks like everything is 1 level deep key=value, that's why I went with stringify.  This might read a little better:

```shell
> Object.entries({z: 123, a: 123, b: "foo", c: "bar baz"}).sort().map(([k,v])=>`${k}="${v}"`).join(' ')
'a="123" b="foo" c="bar baz" z="123"'
```

Basically the point here is:  validation failed.  Here's some context from AJV.  We could add some text such as `please double check your XML structure`?

Reply:
Yep, no problem. On point 4 -- happy to return extra detail from AJV so long as there is a preamble "Unable to validate XML" or something like that.